src/Middleware/RedirectToLogin.php

```diff
@@ -38,10 +38,31 @@ public function __construct(Horde_Registry $registry, ResponseFactoryInterface $
     }
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
+        
         if ($request->getAttribute('HORDE_AUTHENTICATED_USER')) {
             return $handler->handle($request);
         }
-        $redirect = (string)Horde::Url($this->registry->getInitialPage('horde'), true);
+
+        // set baseurl: check if alternative login is set
+        if(isset($GLOBALS['conf']['auth']['alternate_login'])){
+            $baseurl = $GLOBALS['conf']['auth']['alternate_login'];
+        }
+        else{
+            $baseurl = $this->registry->getInitialPage('horde');
```

Comment:
you can use `$this->registry->getServiceLink('login');` to get login url.

Reply:
thanks!